bears/bearController.js

```diff
@@ -1,24 +1,67 @@
-const router = require('express').Router();
+const router = require('express').Router(); // declare that all routes for this address will be found on this router
+
+const Bear = require('./bearModel'); // pull in our Bear model
 
 router
   .route('/')
   .get((req, res) => {
-    res.status(200).json({ route: '/api/bears/' });
+    Bear.find() // This will find ALL resources at that model.
+      .then(bears => {
+        res.status(200).json(bears);
+      })
+      .catch(err => res.status(500).json({ error: 'Error fetching bears' }));
   })
   .post((req, res) => {
-    res.status(201).json({ status: 'please implement POST functionality' });
+    const { species, latinName } = req.body;
+    const newBear = new Bear({ species, latinName });
+    newBear
+      .save() // this will `insert` a document into the Bear collection
+      .then(savedBear => {
+        res.status(201).json(savedBear);
+      })
+      .catch(err => {
+        res.status(422).json({ error: err });
+      });
   });
 
 router
   .route('/:id')
   .get((req, res) => {
-    res.status(200).json({ route: '/api/bears/' + req.params.id });
+    const { id } = req.params;
+    Bear.findById(id) // find a specific resource in a collection by ID
+      .then(foundBear => {
+        res.status(200).json(foundBear);
+      })
+      .catch(err => {
+        res.status(404).json({ error: 'No bear by that id in DB to get' });
+      });
   })
+
   .delete((req, res) => {
-    res.status(200).json({ status: 'please implement DELETE functionality' });
+    const { id } = req.params;
+    Bear.findByIdAndRemove(id)
+      .then(byeBear => {
+          res.status(204).json(beyBear).end();
+      })
+      .catch(err => {
+          res.status(404).json({ error: 'No bear by that id in DB to delete' });
+      })
   })
+  
   .put((req, res) => {
-    res.status(200).json({ status: 'please implement PUT functionality' });
+    const { id } = req.params;
+    const update = req.body;
+    const options = {
```

Comment:
Where are options being used? Is there a reference to this.options somewhere or something?

Reply:
It's a style I acquired from CS9.

    const options = { new: true, };
is used in the next line, as
    Bear.findByIdAndUpdate(id, update, options)

It is identical to the
    Bear.findByIdAndUpdate(id, update, {new: true})